src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -6608,6 +6618,10 @@ uint32 Unit::SpellDamageBonusTaken(Unit* caster, SpellInfo const* spellProto, ui
                 TakenTotalCasterMod += (float((*i)->GetAmount()));
         }
 
+        // Versatility
+        if (damagetype == DOT && GetSpellModOwner())
+            pdamage -= CalculatePct(pdamage, GetSpellModOwner()->GetRatingBonusValue(CR_VERSATILITY_DAMAGE_TAKEN) + GetSpellModOwner()->GetTotalAuraModifier(SPELL_AURA_MOD_VERSATILITY));
```

Comment:
So if you have 100% Versatility.. you will practically take 0 damage?

Reply:
Theoretically, yes.

Reply:
Only 50% of SPELL_AURA_MOD_VERSATILITY for damage reduction...